### ISSUE TITLE
Formatting Input to ONNX Sklearn converter

### ISSUE THREAD

**Issue by idiasuley** [t-wolfeadam/AnyLogic-ONNX-Helper#5]
Hello :) I'm working on embedding a trained RandomForest model into a simulation so i can see how the results of the model influence behavior. I just found the ONNX Helper and so far so awesome. I was hoping for some help in figuring out how to prepare inputs in the simulation model for ONNX. I'm using the sk_learn converter and it expects output in a format that looks like this: initial_type = [('input', FloatTensorType([None, 4]))]. I have tried passing the input from the model as a simple array [4], HashMap (String, int[4]) and Pair (String, int[4]). The first one (simple array) fails and I get an error stating: _Object at index 0 (shape 4) failed to match expected size (input 'X', shape *,4)._ The last two also fail and I get a NullPointerException. When I trace, it's failing at _onnx_wrapper.OnnxHelper.findArrayDims_. **Any help is appreciated, thank you!**

Below is my AnyLogic Attempt
`int[] input0 = new int[]{1, 2, 4, 1};
//Map<String, int[]> inputs = new HashMap<>();
//inputs.put("X", input0);
Pair<String, int[]> inputs = new Pair("X", input0);
traceln(inputs);
//output = onnxHelper.predict(inputs);
int[] output = onnxHelper.predict(int[].class, input0);
traceln(output);`

Here's my ONNX model:

`initial_type = [('float_input', FloatTensorType([None, 4]))]
onx = convert_sklearn(clf, initial_types=initial_type)
with open("rf_test.onnx", "wb") as f:
    f.write(onx.SerializeToString())`

**Comment by shirtandtieler** [comment]
Hi! Your `[None,4]` input shape (referred to as `*,4` in the helper) indicates the first dimension is variable sized. If you're only passing one value, then just wrap it in another array (e.g., `new int[][]{{1,2,4,1}}`). If you're still running into issues, uploading it here would be helpful, if it's not anything confidential.

Also, if the tensor is a float type, you should use `float` instead of `int` (but you can double check what it expects by looking clicking the helper during runtime to see the inspection window).

**Comment by idiasuley** [comment]
Thank you so much for responding! I should have thought of wrapping the array! That resolved the shape problem but I'm having data type issues despite aligning all data types. ONNX doesn't support Int32 so i just changed everything to float but now I'm getting errors stating that there are mismatches with the input as double? I've attached my testing model (my actual model is different but the way it works is the same). Thanks again for helping :)
[Sales Funnel_Test.zip](https://github.com/t-wolfeadam/AnyLogic-ONNX-Helper/files/8982530/Sales.Funnel_Test.zip)

**Comment by shirtandtieler** [comment]
Glad to help! And thank you for attaching the test model, it was helpful to see the full context.

> there are mismatches with the input as double

If you check the inspection window during runtime, you'll see the input needs to be a double. 

![image](https://user-images.githubusercontent.com/1811239/175783380-79f8f261-f843-403f-9456-83c70dbb4104.png)

I confirmed this information is the same as in Python (i.e., not a bug with the helper) by printing out `onx.graph.input` and `onx.graph.output`. So it seems it's an issue with how skl2onnx converts the classifier to an ONNX file. 

Though, regarding parsing the output, there *is* an issue with the helper in that it assumes you requesting a single output means there's only one output in the model (which doesn't work in this case, as there's two outputs). I'll look into adding type detection to the helper 🙂 

In short, with the current ONNX file, you need to make two changes to your AL code:

1. Change the input type to double
2. Use the single-argument form of `predict` (returns a map) and get the labels by querying the input name

i.e.,

```java
double[][] input0 = new double[][]{{1,2,4,1}};

// To get labels and probabilities:
//Map<String,Object> outputs = onnxHelper.predict(input0);
//long[] labels = (long[]) outputs.get("output_label");
//List<Map<Long,Double>> probs = (List) output.get("output_probability");

// To just get labels:
long[] labels= (long[])onnxHelper.predict(input0).get("output_label");
traceln(labels[0]);
```

BTW, ONNX does seem to support int32, as per their [spec doc](https://github.com/onnx/onnx/blob/main/docs/IR.md#tensor-element-types).

**Comment by idiasuley** [comment]
Thank you so much Tyler!!! Worked like a charm after applying to my main model. I really appreciate the help and I appreciate ONNX Helper. I'm excited to run experiments now.
